lib/bolt/cli.rb

```diff
@@ -532,7 +530,10 @@ def in_bolt_compiler(opts = [])
         end
       end
 
-      raise Bolt::CLIError, r.message if r.is_a? StandardError
+      if r.is_a? StandardError
+        r = Bolt::CLIError.new(r.message) unless r.is_a? Bolt::Error
```

Comment:
I don't quite follow this code. Should it just be another rescue inside the block above? It's not clear to me what `e.cause` returns for a `Puppet::PreformattedError`...

Reply:
There are two things going on.
I'm not sure what the goal of returning e.cause is. At least in some cases it's silently squashing all output into a nil return value.

I can move some of this logic up into the compiler block and rely on rescue instead of conditionals which seems better.